### PR TITLE
ui: Side menu redesign

### PR DESCRIPTION
**Pull request by ericholguin**
## Summary

This PR redesigns the side menu to more closely match Roberto's design

Closes: #2541

## Checklist

- [ x ] I have read (or I am familiar with) the [Contribution Guidelines](../docs/CONTRIBUTING.md)
- [ x ] I have tested the changes in this PR
- [ x ] My PR is either small, or I have split it into smaller logical commits that are easier to review
- [ x ] I have added the signoff line to all my commits. See [Signing off your work](../docs/CONTRIBUTING.md#sign-your-work---the-developers-certificate-of-origin)
- [ x ] I have added appropriate changelog entries for the changes in this PR. See [Adding changelog entries](../docs/CONTRIBUTING.md#add-changelog-changed-changelog-fixed-etc)
    - [ ] I do not need to add a changelog entry. Reason: _[Please provide a reason]_
- [ x ] I have added appropriate `Closes:` or `Fixes:` tags in the commit messages wherever applicable, or made sure those are not needed. See [Submitting patches](https://github.com/damus-io/damus/blob/master/docs/CONTRIBUTING.md#submitting-patches)

## Test report

_Please provide a test report for the changes in this PR. You can use the template below, but feel free to modify it as needed._

**Device:** _[Please specify the device you used for testing]_

- iPhone SE (3rd Generation)
- iPhone 12 mini
- iPhone 15 Pro Max

**iOS:** _[Please specify the iOS version you used for testing]_

- 16.4
- 17.0

**Damus:** _[Please specify the Damus version or commit hash you used for testing]_

1.11 (1) 735fa970

**Setup:** _[Please provide a brief description of the setup you used for testing, if applicable]_

Simulator

**Steps:** _[Please provide a list of steps you took to test the changes in this PR]_

Verified all side menu functionality remained the same and the UX did not change

**Results:**
- [ x ] PASS
- [ ] Partial PASS
  - Details: _[Please provide details of the partial pass]_

## Other notes

_[Please provide any other information that you think is relevant to this PR.]_

Changelog-Changed: Changed side menu design

### CODE REVIEW

File: damus/Views/PubkeyView.swift [hunk 1]
```diff
@@ -72,7 +74,7 @@ struct PubkeyView: View {
                                 .resizable()
                                 .contentShape(Rectangle())
                                 .foregroundColor(colorScheme == .light ? DamusColors.darkGrey : DamusColors.lightGrey)
-                                .frame(width: 20, height: 20)
+                                .frame(width: 15, height: 15)
```
Comment (danieldaquino) [inline damus/Views/PubkeyView.swift:77]: These sizing changes affect other instances of `PubkeyView`, such as in the profile view and the profile action sheet. This smaller size makes sense on the side menu, but maybe not in the other views. Was this intended?

If this is not intended, perhaps we can implement an `OptionSet` argument on `PubkeyView` to allow the caller to choose how to display this? (Similar to how we handle different display styles on `EventView` by using `EventViewOptions`)

**Review by danieldaquino — COMMENTED** [review]
New side menu looks great! Just added one comment regarding the pubkey view sizing changes affecting its other usages, but otherwise LGTM!

**Comment by jb55** [comment]
needs rebase

**Comment by alltheseas** [comment]
moving from sprint 25 to 26 (starting oct 14 2024)

**Comment by ericholguin** [comment]
> needs rebase

Done

**Review by danieldaquino — APPROVED** [review]
Code and functionality LGTM! Approved.

Passing to @jb55 for final review

**Pull request merged**